src/backend/stream.rs

```diff
@@ -355,6 +355,13 @@ impl<'ctx> PulseStream<'ctx> {
                         if got < 0 || got as usize != read_frames {
                             let _ = s.cancel_write();
                             stm.shutdown = true;
+                            unsafe {
+                                stm.state_callback.unwrap()(
+                                    stm as *mut _ as *mut _,
+                                    stm.user_ptr,
+                                    ffi::CUBEB_STATE_ERROR,
+                                );
+                            }
```

Comment:
```suggestion
                            if got < 0 {
                                unsafe {
                                    stm.state_callback.unwrap()(
                                        stm as *mut _ as *mut _,
                                        stm.user_ptr,
                                        ffi::CUBEB_STATE_ERROR,
                                    );
                                }
                            }
```

From https://github.com/mozilla/cubeb/blob/d97fea4c9015b5d79e11ad68e9cdac610c27ca07/include/cubeb/cubeb.h#L418-L420:
> If the stream is input only, then returning nframes indicates data has been read. In this case, a value less than nframes will result in the stream being stopped.

This will fire error callback when user return a less than nframes value (`got as usize != read_frames`) as well. I think we should fire error callback only if `got < 0`, as what cubeb_wasapi.cpp does?

Reply:
Good catch, thanks! I've fixed this in cubeb_pulse.c also.